storage-node/src/services/logger.ts

```diff
@@ -210,6 +188,24 @@ function createElasticTransport(logSource: string, elasticSearchEndpoint: string
     index: 'storage-node',
     format: ecsformat(),
     source: logSource,
+    retryLimit: 10,
   }
   return new ElasticsearchTransport(esTransportOpts)
 }
+
+/**
+ * Creates winston logger file transport.
+ *
+ * @param fileName - log file name.
+ * @returns winston file transport
+ */
+function createFileTransport(filename: string): winston.transport {
+  const options = {
+    filename,
+    maxsize: 50000000, // 50 MB
+    maxFiles: 3,
+    level: 'debug',
+    format: ecsformat(),
+  }
+  return new winston.transports.File(options)
```

Comment:
For distributor node I used [`winston-daily-rotate-file` transport](https://github.com/Lezek123/substrate-runtime-joystream/blob/distributor-node/distributor-node/src/services/logging/LoggingService.ts#L102), because I noticed a strange behavior when using the default `winston.transports.File` with both `maxsize` and `maxFiles` set.

To ilustrate this: I changed the `maxsize` for Colossus to `10KB`, gathered some logs and then ran `distributor-node dev:batchUpload` script to execute 200 uploads at once. The result is as follows:
![image](https://user-images.githubusercontent.com/19978260/146975837-f892c784-aad4-4a33-b32b-8a6891b6ee67.png)
First there is `log.txt` (generated before I decreased the maxsize) and `log1.txt`. You would then expect `log.txt`, `log1.txt` and `log2.txt`, but next time I'm checking the logs (while the `batchUpload` is executing), there is only `log1.txt`. Then when the `batchUpload` finishes executing we suddenly have `log275.txt` out of nowhere.

This matches the experience I had when testing distributor node file logging, I think `winston.transports.File` is just not a reliable transport.

Reply:
Ok. Copied most part of the implementation from Argus (https://github.com/Joystream/joystream/pull/2939/commits/2f527e86e1c34fa5a2fcd610cb0bdf9c48e0b8db)